src/main/java/eu/adlogix/appnexus/oas/client/service/CampaignService.java

```diff
@@ -0,0 +1,111 @@
+package eu.adlogix.appnexus.oas.client.service;
+
+import static eu.adlogix.appnexus.oas.client.utils.ValidatorUtils.checkNotEmpty;
+import static eu.adlogix.appnexus.oas.client.utils.ValidatorUtils.checkNotNull;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import eu.adlogix.appnexus.oas.client.domain.Campaign;
+import eu.adlogix.appnexus.oas.client.domain.StatefulDomainManager;
+import eu.adlogix.appnexus.oas.client.domain.enums.BillTo;
+import eu.adlogix.appnexus.oas.client.domain.enums.CampaignType;
+import eu.adlogix.appnexus.oas.client.parser.XmlToCampaignParser;
+import eu.adlogix.appnexus.oas.client.transform.CampaignCreateParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.transform.CampaignParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.transform.CampaignUpdateParameterMapTransformer;
+import eu.adlogix.appnexus.oas.client.xml.ResponseParser;
+import eu.adlogix.appnexus.oas.client.xml.XmlRequestGenerator;
+
+/**
+ * Service Class which provides functions for all Campaign related operations
+ * 
+ */
+public class CampaignService extends AbstractOasService {
```

Comment:
Yes I think it is better to have interfaces for the services as we could have different implementations of the services. (may be in the future we will have a different way of accessing the oas services ). 
What should we name the interface and the implementation class?